src/libvirtApi/storagePool.js

```diff
@@ -68,79 +68,75 @@ export function storagePoolDeactivate({ connectionName, objPath }) {
+
+        const [resultProps] = await call(connectionName, objPath, 'org.freedesktop.DBus.Properties', 'GetAll', ['org.libvirt.StoragePool'], { timeout, type: 's' });
+        /* Sometimes not all properties are returned; for example when some storage got deleted while part
+            * of the properties got fetched from libvirt. Make sure that there is check before reading the attributes.
+            */
+        if ("Active" in resultProps)
+            props.active = resultProps.Active.v.v;
+        if ("Persistent" in resultProps)
+            props.persistent = resultProps.Persistent.v.v;
+        if ("Autostart" in resultProps)
+            props.autostart = resultProps.Autostart.v.v;
+
+        props.volumes = [];
+        if (props.active) {
+            const volumes = await storageVolumeGetAll({ connectionName, poolName: dumpxmlParams.name });
+            props.volumes = volumes;
+        }
+        store.dispatch(updateOrAddStoragePool(Object.assign({}, dumpxmlParams, props), updateOnly));
+    } catch (ex) {
+        console.warn('GET_STORAGE_POOL action failed for path', objPath, ex.toString());
+    }
 }
 
-export function storagePoolGetAll({
+export async function storagePoolGetAll({
     connectionName,
 }) {
-    return call(connectionName, '/org/libvirt/QEMU', 'org.libvirt.Connect', 'ListStoragePools', [0], { timeout, type: 'u' })
-            .then(objPaths => {
-                return Promise.all(objPaths[0].map(path => {
-                    return call(connectionName, path, 'org.freedesktop.DBus.Properties', 'Get', ['org.libvirt.StoragePool', 'Active'], { timeout, type: 'ss' })
-                            .then(active => {
-                                if (active[0].v)
-                                    return storagePoolRefresh({ connectionName, objPath: path });
-                                else
-                                    return storagePoolGet({ connectionName, id: path });
-                            });
-                }));
-            })
-            .catch(ex => {
-                console.warn('GET_ALL_STORAGE_POOLS action failed:', ex.toString());
-                return Promise.reject(ex);
-            });
+    try {
+        const [objPaths] = await call(connectionName, '/org/libvirt/QEMU', 'org.libvirt.Connect', 'ListStoragePools', [0],
+                                      { timeout, type: 'u' });
+        return await Promise.all(objPaths.map(async path => {
+            const [active] = await call(connectionName, path, 'org.freedesktop.DBus.Properties', 'Get',
+                                        ['org.libvirt.StoragePool', 'Active'], { timeout, type: 'ss' });
+            if (active.v)
+                return storagePoolRefresh({ connectionName, objPath: path });
+            else
+                return storagePoolGet({ connectionName, id: path });
+        }));
+    } catch (ex) {
+        console.warn('GET_ALL_STORAGE_POOLS action failed:', ex.toString());
+        throw ex;
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1400-20240125-053137-bfbdf98b-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)